lib/ansible/module_utils/dimensiondata.py

```diff
@@ -61,43 +61,10 @@ def check_libcloud_or_fail():
         raise LibcloudNotFound("apache-libcloud is required.")
 
 
-def get_configured_credentials(module):
+def get_credentials(module):
     """
-    Attempt to retrieve credentials from the module parameters.
-    If 'mcp_user' is present then 'mcp_password' is also required.
-
-    Returns None if 'mcp_user' is not present.
-
-    If successful, returns a dict with 'user_id' and 'key'
-    representing the MCP username and password.
-    """
-
-    if not HAS_LIBCLOUD:
-        module.fail_json(msg='libcloud is required for this module.')
-
-        return None
-
-    if 'mcp_user' not in module.params:
-        return None
-
-    if 'mcp_password' not in module.params:
-        module.fail_json(
-            '"mcp_user" parameter was specified, but not "mcp_password" ' +
-            '(either both must be specified, or neither).'
-        )
-
-        return None
-
-    return dict(
-        user_id=module.params['mcp_user'],
-        key=module.params['mcp_password']
-    )
-
-
-def get_credentials():
-    """
-    This method will get user_id and key from module configuration, environment, or dotfile.
-    Environment takes priority.
+    Get user_id and key from module configuration, environment, or dotfile.
+    Order of priority is environment, dotfile, module.
```

Comment:
Sorry, I think the current logic will only fall back to module if environment variable has not been specified: https://github.com/ansible/ansible/pull/17604/files/6c412fbeefa07e4296b33ec7544c5b1a69a6dcd8..85164a272274514877a6369877e851461a5acf0b#diff-0b1c579a3ccbeed5e582e7693e046cb6R112

I'll change the order, though, to put module first. Give me 5 mins.